### PR TITLE
feat: support port ranges when starting a container (#3204)

### LINKED ISSUE #3204: Adding support for range port mappings
### Is your enhancement related to a problem? Please describe

Not a problem at all. I couldn't define in podman-desktop for a container a range port mapping.

Currently is only possible to add a single port mapping. This is possible to add multiple times but when we try in podman CLI -p 30000-32000:30000-32000 it is possible to define ranges but not in podman-desktop.

### Describe the solution you'd like

I would like, near to the current portmappings settings, the posibillity to add port mapping ranges.

### Describe alternatives you've considered

I tried also to add in the single port mapping but with the range values like 30000-32000 for the host and the container like 30000-32000 but the sintax was not working.

Maybe instead to create new fields in the settings for the container in order to support port mapping ranges, it will be helpfull if the sintax for ranges will be possible to set.  

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

it adds the logic to handle port ranges in the form of <port>-<port>

### Screenshot/screencast of this PR

range_port

### What issues does this PR fix or reference?

it resolves #3204 

### How to test this PR?

1. start a container by mapping a range of ports and check that it actually worked by inspecting it

### CODE REVIEW

File: packages/main/src/plugin/util/port.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+});
+
+test('return port range starting with port 9001 if 9000 is busy', async () => {
+  const server = net.createServer();
+  server.listen(9000);
+  const range = await port.getFreePortRange(3);
+
+  expect(range).toBe('9001-9003');
+  server.close();
+});
```
Comment (benoitf) [inline packages/main/src/plugin/util/port.spec.ts]: how will it work if I have something running on my side on the port 9000 ? (I know I saw some apps listening on 9000 port)

should the test should pickup a free port/range before ?

Comment (lstocchi) [reply packages/main/src/plugin/util/port.spec.ts]: you're right, so i guess it only makes sense to check if the range is valid and the start/end values are valid numbers

Comment (benoitf) [reply packages/main/src/plugin/util/port.spec.ts]: for the first one, yes probably you ask for a range and then create server on it then it ensures it gave you free ports

And for the second probably you ask for a range, get the ports then use the first port and ask again and see that it's shifted

Comment (lstocchi) [reply packages/main/src/plugin/util/port.spec.ts]: @benoitf updated

**Review by benoitf — APPROVED** [review]

**Pull request merged**